File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<bits/stdc++.h>

using namespace std;

bool solution(string s) {
    bool answer = true;
    
    stack<char> sta;
    
    for(auto c : s) {
        if(c == '(') sta.push(c);
        else {
            if(sta.empty()) {
                answer = false;
                break;
            }
            else if (sta.top() == '(') sta.pop();
            else {
                answer = false;
                break;
            }
        }

    }

    if(!sta.empty()) answer = false;
    
    return answer;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    isRight = True
    stack = []
    for i in s:
        if i == '(':
            stack.append(i)
        else:
            if len(stack) == 0:
                return False
            top = stack[len(stack) - 1]
            if (top == '('):
                stack.pop()
            else:
                return False
            

    if stack != []:
        isRight = False

    return isRight